datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -390,8 +406,13 @@ fn build_statistics_record_batch<S: PruningStatistics>(
     }
 
     let schema = Arc::new(Schema::new(fields));
-    RecordBatch::try_new(schema, arrays)
-        .map_err(|err| DataFusionError::Plan(err.to_string()))
+    // provide the count in case there were no needed statistics
```

Comment:
Previously, if the pruning predicate did not call for any columns (b/c it was "true" for example) this would error because in earlier versions of `arrow-rs` `RecordBatch`es could not have `0` columns -- now that they can, we use that feature to build a 0 column RecordBatch when the predicate has no columns that can be transformed